yt/data_objects/particle_trajectories.py

```diff
@@ -237,13 +245,15 @@ def _get_data(self, fields):
             field for field in missing_fields if field not in self.particle_fields
         ]
         step = int(0)
+        fields_str = ", ".join(str(f) for f in missing_fields)
         pbar = get_pbar(
-            f"Generating [{', '.join(missing_fields)}] fields in trajectories",
-            self.num_steps,
+            f"Generating [{fields_str}] fields in trajectories", self.num_steps,
         )
         my_storage = {}
 
-        for i, (sto, ds) in enumerate(self.data_series.piter(storage=my_storage)):
+        for i, (sto, ds) in enumerate(
+            self.data_series.piter(storage=my_storage, dynamic=False)
+        ):
```

Comment:
I think the comment you wrote above about the use of `dynamic=False` might be useful here too.
One way to avoid duplication would be to add this comment to the method's docstring.

Reply:
Should be good now!